CHANGELOG.md

```diff
@@ -12,6 +12,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Documentation
 ### Maintenance
 ### Refactoring
+- Isolate KNNQueryBuilder creation in NeuralKNNQueryBuilder ([#1183](https://github.com/opensearch-project/neural-search/pull/1183))
```

Comment:
suggest slight change in description: "Encapsulate KNNQueryBuilder creation within NeuralKNNQueryBuilder", if you're ok with it please also change the PR header

Reply:
Updated in the changeLog and PR description.